tls/s2n_tls_parameters.h

```diff
@@ -221,8 +225,12 @@
 #define S2N_DEFAULT_RECORD_LENGTH 8092
 #define S2N_DEFAULT_FRAGMENT_LENGTH (S2N_DEFAULT_RECORD_LENGTH - S2N_TLS_RECORD_HEADER_LENGTH)
 
-#define S2N_LARGE_RECORD_LENGTH S2N_TLS_MAXIMUM_RECORD_LENGTH
+/* S2N_LARGE_RECORD_LENGTH is used for output initializing buffers, we use the largest
+ * possible value of all supported protocols to avoid branching at runtime
+ */
+#define S2N_LARGE_RECORD_LENGTH S2N_TLS13_MAXIMUM_RECORD_LENGTH
```

Comment:
actually TLS 1.2 allows for the biggest maximum record length, which is 2^14 + 2048 octets (https://tools.ietf.org/html/rfc5246#section-6.2.3)

These values are also summarized in the first section of https://tools.ietf.org/html/draft-ietf-tls-record-limit-03

Reply:
as above.